### ISSUE TITLE
Rename sqrt to square_root

### ISSUE THREAD

**Issue by wbhart** [Nemocas/AbstractAlgebra.jl#555]
Currently sqrt(2) returns 1.404.... in Julia. But this is not useful for an algebraic system. Therefore, after some discussion with @fieker, we have decided to rename sqrt to square_root.

It should raise an exception if the square root is not exact. Of course we can still have isqrt for floor of the square root.

I think either the polynomial or series sqrt currently returns a square root even if there is a remainder, so we would need to have the same behaviour there. But that behaviour is relied upon for some other algorithms (I think in either the LaurentSeries or PuiseuxSeries or Poly modules), so we may need to rename the current function to something else, e.g. square_root_with_remainder or something like that.

**Comment by wbhart** [comment]
After spending some considerable time trying to remember the discussion we previously had on this, I now believe I recall the details.

@fieker made the point that whilst sqrt may sound natural to some ears and to people who are used to this from their background, it isn't what others would go looking for. Also, I believe he made the point that eventually you run out of four letter abbreviations and that elsewhere we have started using longer names (denominator instead of det, etc.) with underscores, instead of following Julia abbreviations.

I would counter that Julia users will go looking for isqrt and sqrt. I don't believe it is appropriate to overload isqrt (integer square root) for all our square root purposes, therefore the solution most compatible with the language is to define sqrt internally in Nemo/AbstractAlgebra to do what we want.

Oscar can still call it whatever is deemed appropriate by a function that calls Nemo.sqrt.

However, if @fieker 's recollection of the discussion is the same as mine, and he still strongly prefers square_root, I am not opposed. But it's not easy to come up with a similar long name for inv. Inverse sounds contrived and invert is not a noun.

Anyhow, once we have a decision (now that some time has passed to see how this sits with us), it should be incorporated into the suggestions at https://github.com/Nemocas/Nemo.jl/issues/862 before finalising the interface there.

It would also be good to hear the perspective of @rfourquet @thofma and @tthsqe12 on this ticket. Perhaps @fingolfin and others would like to weigh in. We simply need to find something everyone can live with. I can live with either suggestion above, (but obviously prefer sqrt for the reasons I gave above).

https://github.com/Nemocas/AbstractAlgebra.jl/issues/556 is also tangentially relevant here, though I strongly believe that should be confined to integer square root only.
